[HttpCache] Add limitations to example of using vary by cookie header (#1169)
ezsystems/developer-documentation: docs/guide/http_cache.md

1\. ESI + Vary by Cookie:

- As default VCL strips everything but session cookie this means this will effectively be cached "per user".

>     - With Symfony Proxy you need to implement cookie stripping using web server (Nginx/Apache) configuration

Reviewer: ```suggestion
    - With Symfony Proxy you must modify web server (Nginx/Apache) configuration to implement cookie stripping.
```

Author: +1 from my side

>     - If you are on single server setup with neither Varnish nor Fastly, you can do the same cookie logic on the Web server instead.

Reviewer 2: ```suggestion
    - If you are on single-server setup with neither Varnish nor Fastly, you can use the same cookie logic on the Web server instead.
```

>     - If you are on single server setup with neither Varnish nor Fastly, you can do the same cookie logic on the Web server instead.

Reviewer 3: I am a bit fuzzed by this sentence... We are in chapter "ESI + Vary by Cookie", so you obviously are using Varnish or Fastly. And if you are not, you're likely using the Symfony proxy so you cannot simply do logic on the web server... ?

Author: ![Screenshot 2020-10-13 at 09 42 45 ](https://user-images.githubusercontent.com/289757/95831308-292c5400-0d39-11eb-8592-5f8f513c10d5.png)

This section is for all supported proxies, except proposals below which depends on being able to modify VCL logic to happen.

>     - If you are on single-server setup with neither Varnish nor Fastly, you can use the same cookie logic on the Web server instead.

Author: Possible suggestion to simplify this sentence:
```suggestion
    - With Symfony Proxy you'll need to implement cookie stripping on Web server _(Nginx/Apache)_ configuration
```
Notes:
- Possible extension to this sentence `, in order to avoid this varying on all possible analytics cookies and so on.`
- This is not possible on platform.sh, but we ship with Varnish by default there.
- It's technical possible to implement this by adapting `src/CacheKernel.php`, but that is not advisable in terms of upgradability.

>     - If you are on single-server setup with neither Varnish nor Fastly, you can use the same cookie logic on the Web server instead.

Reviewer 2: ```suggestion
    - With Symfony Proxy you need to implement cookie stripping using web server (Nginx/Apache) configuration
```
